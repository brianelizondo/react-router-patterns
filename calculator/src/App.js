import {React, useState } from 'react';
import { Routes, Route } from "react-router-dom";
import './App.css';

import Add from "./Add";
import Subtract from "./Subtract";
import Multiply from "./Multiply";
import Divide from "./Divide";

import Calculator from './Calculator';

function App() {
    return (
        // REACT ROUTER CALCULATOR WITH DIFFERENTS COMPONENTS
        // <div className="App">
        //     <Routes>
        //         <Route path="/add/:a/:b" element={ <Add /> } />
        //         <Route path="/subtract/:a/:b" element={ <Subtract /> } />
        //         <Route path="/multiply/:a/:b" element={ <Multiply /> } />
        //         <Route path="/divide/:a/:b" element={ <Divide /> } />
        //     </Routes>
        // </div>

        <div className="App">
            <Routes>
                <Route path="/:operation/:a/:b" element={ <Calculator /> } />
            </Routes>
        </div>
    );
}

export default App;
