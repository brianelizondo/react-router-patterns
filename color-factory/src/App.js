import {React, useState } from 'react';
import { Routes, Route, Navigate } from "react-router-dom";
import './App.css';

import ColorList from "./ColorList";
import ColorDetails from "./ColorDetails";
import ColorNew from "./ColorNew";

function App() {
    const INITIAL_STATE = [];
    const [colors, setColors] = useState(INITIAL_STATE);

    const addColor = newColor => {
        setColors(colors => [newColor, ...colors]);
    };

    return (
        <div className="App">
            <Routes>
                <Route path="/colors/new" element={ <ColorNew addColor={addColor} /> } />
                <Route path="/colors/:color" element={ <ColorDetails colors={colors} /> } />
                <Route path="/colors" element={ <ColorList colors={colors} /> } />
            </Routes>
        </div>
    );
}

export default App;
