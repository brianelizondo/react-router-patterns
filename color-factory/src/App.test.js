import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import App from './App';

// Smoke Tests
it("'App' component renders without crashing", () => {
    render(<MemoryRouter><App /></MemoryRouter>);
});

// Snapshot Tests
it("'App' matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

// Others test
test("/colors route", () => {
    const { getByText } = render((
        <MemoryRouter initialEntries={["/colors"]}>
            <App />
        </MemoryRouter>
    ));
    
    expect(getByText("Welcome to Color Factory")).toBeInTheDocument();
});

test("/colors/new route", () => {
    const { getByText } = render((
        <MemoryRouter initialEntries={["/colors/new"]}>
            <App />
        </MemoryRouter>
    ));
    
    expect(getByText("Add New Color")).toBeInTheDocument();
});
