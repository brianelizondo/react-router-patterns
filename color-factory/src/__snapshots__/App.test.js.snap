// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'App' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="ColorList"
    >
      <div
        class="ColorList-title"
      >
        <div>
          Welcome to Color Factory
        </div>
        <div>
          <a
            href="/colors/new"
          >
            Add a color
          </a>
        </div>
      </div>
      <div
        class="ColorList-colors"
      >
        <div>
          <h4>
            Please select a color:
          </h4>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
